src/execution_plan/ops/op_value_hash_join.c

```diff
@@ -90,7 +90,9 @@ static bool _set_intersection_idx(OpValueHashJoin *op, SIValue v) {
 	// Make sure value was found.
 	Record r = op->cached_records[leftmost_idx];
 	SIValue x = Record_Get(r, op->join_value_rec_idx);
-	if(SIValue_Compare(x, v, NULL) != 0) return false; // Value wasn't found.
+	int disjointOrNull = 0;
+	// Return false if the value wasn't found or evaluated to NULL.
+	if(SIValue_Compare(x, v, &disjointOrNull) != 0 || disjointOrNull == COMPARED_NULL) return false;
```

Comment:
This change is not strictly necessary, as LHS Records that attempt to join NULL values will have already been discarded by this point. I consider it a sensible precaution, though.

Reply:
@jeffreylovitz @DvirDukhan 
Generally speaking, looking at `SIValue_Compare` when comparing two NULL values, the function returns 0, indicating they are equal, but will set `disjointOrNull` to `COMPARED_NULL`, this is problematic as `disjointOrNull` isn't mandatory argument, I think it would be better to return none zero incase we're comparing two null values.